### PR TITLE
[NCL-5350] [NCL-5272]: WIP - migrate cleaner to work with Orch 2.0

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

File: src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java [hunk 1]
```diff
@@ -265,41 +262,43 @@ boolean shouldClean(String groupName, FailedBuildsCleanerSession session) throws
...
-                logger.debug("Attempting to find build record by id {}", id);
-                BuildRecordSingleton singleton = null;
-                try {
-                    singleton = buildService.getSpecific(id);
-                } catch (WebApplicationException ex) {
-                    int status = ex.getResponse().getStatus();
-                    if (status == Response.Status.NOT_FOUND.getStatusCode()) {
+
+        Matcher matcher = buildNumPattern.matcher(buildContentId);
+        String id = matcher.group(1);
```
[the remark this answers is not in the document]
Comment (pkocandr) [reply src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java]: And also this was supposed to be done after a build cannot be found by buildContentId. I believe we still have some edge cases when buildContentId is left empty, so search by parsed ID is a fallback in case it can be parsed. This should be moved to line 282.

[the remark this answers is not in the document]
Comment (jbartece) [reply src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java]: OK, I missed that. Moved back

File: src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java [hunk 2]
```diff
@@ -68,13 +70,15 @@
...
     static {
-        failedStatuses = new ArrayList<>(2);
-        failedStatuses.add(BuildCoordinationStatus.CANCELLED); // added for old data, now cancellation cleans up the Indy data
-        failedStatuses.add(BuildCoordinationStatus.DONE_WITH_ERRORS);
-        failedStatuses.add(BuildCoordinationStatus.SYSTEM_ERROR);
+        failedStatuses = new ArrayList<>(5);
+        failedStatuses.add(BuildStatus.CANCELLED); // added for old data, now cancellation cleans up the Indy data
+        failedStatuses.add(BuildStatus.SYSTEM_ERROR);
+        failedStatuses.add(BuildStatus.FAILED);
+        failedStatuses.add(BuildStatus.REJECTED_FAILED_DEPENDENCIES);
```
Comment (pkocandr) [inline src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java]: Hmm, why were these added? The rejected builds were not started, so they did not create any repos in Indy. It doesn't make sense to have them here.

Comment (jbartece) [reply src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java]: Good point. You are right. Removing them.
